src/pages/api/requestContent.ts

```diff
@@ -1,7 +1,28 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
 import axios from 'axios';
 
+import rateLimit from '../../utils/rate-limit';
+
+const limiter = rateLimit({
+    max: 600, // cache limit of 600 per 30 second period.
+    ttl: 30 * 1000,
+});
+
+const MAX_CONTENT_REQUEST_PER_PERIOD = 10;
+
 export default async (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+        await limiter.check(
+            res,
+            MAX_CONTENT_REQUEST_PER_PERIOD,
+            req.socket.remoteAddress || ''
+        );
+    } catch {
+        return res.status(500).json({
+            error: { message: 'Something went wrong' },
+        });
+    }
```

Comment:
Wondering if this should be moved into a reusable helper function to be imported in all three cases

Reply:
So I was thinking about doing that, but I kinda wanted to leave it up to the endpoint to determine what the max-per-period requests are and what to do when that max has been reached. We might want to treat different endpoints differently in the future, so I figured I'd try to follow that pattern.